discovery/validation_barrier.go

```diff
@@ -103,7 +104,24 @@ func (v *ValidationBarrier) InitJobDependencies(job interface{}) (JobID,
 	// channel to be closed.
 	select {
 	case <-v.validationSemaphore:
+
 	case <-v.quit:
+
+	default:
+		// We could not get a slot immediately, so we record the time
+		// here how long we wait.
+		waitStart := time.Now()
+		select {
+		case <-v.validationSemaphore:
```

Comment:
Tho a bit hard, but today you can already check for the time it takes if using `trace`. Plus I think we wanna sunset validation barrier tho.

Reply:
ok cool then let's not do it.